Core/Src/main.c

```diff
@@ -75,6 +89,18 @@ int main(void)
 
   /* USER CODE BEGIN Init */
 
+  //allocate memory for private variables
+
+  char uart_buffer[50];
+  int uart_buffer_length;
+
+  uint8_t spi_input_buffer[3]; //3 bytes from MCP3004 ADC
+  uin16_t spi_input_value;
+  uint8_t spi_output_buffer[3] = {MCP3004_START, MCP3004_SINGLEENDED_CH0, 0b00000000};
+
+  uint16_t pwm_compare_register_counts;
+  uint16_t pwm_counter_counts = ((uint16_t) htim1.Init.Period) + 1; // PWM counter period "60000 - 1" => 60000 counts. uint32_t cast as uint16_t
```

Comment:
will `pwm_counter_counts` ever change, if not consider making it `const`. This is also a readability thing to let the reader of the code understand the purpose of the value.